### PR TITLE
Update codeowners

### PR DESCRIPTION
**Pull request by brian-avery**
Until we have a better idea of who should be codeowners here: 

* Make TOC own the root of this repo
* Make someone from the maintainers group approve changes for feature promotion

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1 +1,4 @@
 /features.yaml @linsun @dcberg @louiscryan @smawson @duderino @nrjpoddar
+. @linsun @dcberg @louiscryan @smawson @duderino @nrjpoddar
+/features @istio/maintainers
```
Comment (howardjohn) [inline CODEOWNERS:3]: I think this will also request review from everyone

Comment (brian-avery) [reply CODEOWNERS:3]: I think that's ok. Theoretically this repo is fairly low traffic. I think TOC should be the approvers for the top level. For the features directory, I think it's ok for any maintainer in any work group to be able to approve a change. Unfortunately, I don't think GitHub gives us the flexibility to give someone approval rights without also tagging them as a reviewer.

**Comment by brian-avery** [comment]
@googlebot I accept

**Review by linsun — APPROVED** [review]
let us experiment and adjust approvers as needed.

**Pull request merged**